vmm/src/cpu.rs

```diff
@@ -594,6 +598,34 @@ impl CpuManager {
             )
             .map_err(Error::CommonCpuId)?
         };
+        #[cfg(all(feature = "amx", target_arch = "x86_64"))]
+        // If we have a vm_config then tdx_or_kernel_present is true at this point
+        if config.features.amx {
+            const ARCH_GET_XCOMP_GUEST_PERM: usize = 0x1024;
+            const ARCH_REQ_XCOMP_GUEST_PERM: usize = 0x1025;
+            const XFEATURE_XTILEDATA: usize = 18;
+            const XFEATURE_XTILEDATA_MASK: usize = 1 << XFEATURE_XTILEDATA;
+
+            let amx_tile = unsafe {
+                libc::syscall(
+                    libc::SYS_arch_prctl,
+                    ARCH_REQ_XCOMP_GUEST_PERM,
+                    XFEATURE_XTILEDATA,
+                )
+            };
+
+            if amx_tile != 0 {
+                return Err(Error::AmxEnable(anyhow!("Guest AMX usage not supported")));
+            } else {
+                let mask: usize = 0;
+                let result = unsafe {
+                    libc::syscall(libc::SYS_arch_prctl, ARCH_GET_XCOMP_GUEST_PERM, &mask)
```

Comment:
Don't you need to provide a mutable reference to `mask`? I guess not because this is `unsafe`.

Reply:
Yep, added that to the unsafe comment.